ddtrace/opentelemetry/span.py

```diff
@@ -0,0 +1,125 @@
+        set_status_on_exception=True,  # type: Optional[bool]
+    ):
+        # (...) -> None
+        if start_time is not None:
+            datadog_span.start = start_time
+
+        # Only set if it isn't "internal" to save on bytes
+        if kind != SpanKind.INTERNAL:
+            # TODO: Validate/clean this?
+            datadog_span.set_tag_str("span.kind", kind.name.lower())
+
+        # TODO: Are there other default tags we want to set?
+
+        self._span = datadog_span
+        self._record_execption = record_exception
+        self._set_status_on_exception = set_status_on_exception
+
+        if attributes:
+            self.set_attributes(attributes)
+
+    def end(self, end_time=None):
+        # type: (Optional[int]) -> None
+        """Ends a datadog span"""
+        self._span.finish(finish_time=end_time)
+
+    def get_span_context(self):
+        # type: () -> SpanContext
+        """Returns an opentelemetry SpanContext using values from the active datadog span"""
+        # TODO: This is the bare minimum to get current context, but what about tracestate and flags?
+        return SpanContext(self._span.trace_id, self._span.span_id, False)
+
+    def set_attributes(self, attributes):
+        # type: (Dict[str, AttributeValue]) -> None
+        """Sets attributes/tags"""
+        for k, v in attributes.items():
+            self.set_attribute(k, v)
+
+    def set_attribute(self, key, value):
+        # type: (str, AttributeValue) -> None
+        """Sets an attribute (aka tags) on the internal datadog span"""
+        if isinstance(value, float) or isinstance(value, int):
+            self._span.set_metric(key, value)
+        else:
+            self._span.set_tag_str(key, str(value))
+
+    def add_event(self, name, attributes=None, timestamp=None):
+        # type: (str, Attributes, Optional[int]) -> None
+        """NOOP - events are not supported in the datadog span API"""
+        return
+
+    def update_name(self, name):
+        # type: (str) -> None
+        """Updates the operation name on the internal datadog span."""
+        self._span.name = name
+
+    def is_recording(self) -> bool:
+        """Returns True if attributes can be set on a span"""
+        # FIXME: This should return false if the Span is not active
+        # Use opentelemetry.trace.get_current_span() to check if span is active
+        return not self._span.finished
+
+    def set_status(self, status, description=None):
+        # type: (Union[Status, StatusCode], Optional[str]) -> None
+        """Sets the Status of the datadog span if status code is Error.
+        StatusCode.Unset is not supported. By default datadog spans have the status OK.
+        """
+        if isinstance(status, Status) and status.status_code is StatusCode.Error:
+            self._span.set_exc_info(Exception, status.description or "", "")
+        elif status is StatusCode.Error:
+            self._span.set_exc_info(Exception, description or "", "")
```

Comment:
Should this method be a noop? Should we only update the status of a datadog span if an exception is raised